### ISSUE TITLE
Proxied Angular tries to get assets/artifacts through HTTP even though the browser navigated via HTTPS

### ISSUE THREAD

**Issue by lukaszpochron** [angular/angular#42772]
# Bug Report

### Affected Package
@angular-devkit/build-angular@12.1.0

### Is this a regression?

Yes, the issue was caused by migration from Angular 7. I apologise for not having more precise information about the exact version causing regression. Will try to provide if it turns out to be necessary.

### Description
While:
1. served by .NET inside a Docker container with HTTP,
2. ran through a Nginx reverse proxy,
3. and navigated with the browser via HTTPS,
Angular is trying to fetch assets and all .js files through HTTP, which is causing a Mixed Media error in the browser:

## Minimal Reproduction
I don't believe I can provide other reproduction than: Angular application serving HTTP, running through a Nginx reverse proxy providing HTTPS connection - more detailed setup in Description above.

## Exception or Error
<pre><code>Mixed Content: The page was loaded over HTTPS, but requested an insecure X. This request has been blocked; the content must be served over HTTPS.</code></pre>

## Your Environment

**Angular Version:**
<pre><code>
Angular: 12.0.5
... animations, cdk, cli, common, compiler, compiler-cli, core
... forms, language-service, localize, platform-browser
... platform-browser-dynamic, platform-server, router

Package                                    Version
--------------------------------------------------------------------
@angular-devkit/architect                  0.1201.1
@angular-devkit/build-angular              12.1.1
@angular-devkit/core                       12.1.1
@angular-devkit/schematics                 12.0.5
@nguniversal/module-map-ngfactory-loader   6.1.0
@schematics/angular                        12.0.5
rxjs                                       6.6.7
typescript                                 4.2.4

</code></pre>

**Workaround**
This issue can be temporarily fixed by adding the below in src/index.html:
`<meta http-equiv="Content-Security-Policy" content="upgrade-insecure-requests">`

I think this is not a bad fix, although it should not be required. It has to be noted that there's good browser compatibility with the exception of the beloved Internet Explorer.

**Comment by petebacondarwin** [comment]
When you say:

> Angular is trying to fetch assets and all .js files through HTTP

What aspect of Angular is doing the fetching? I don't remember Angular normally making XHR requests for JS files.
Are you using SystemJS or something to load the scripts?

**Comment by lukaszpochron** [comment]
Of course, sorry for the wording. These are not `XHR `requests, but `script`, or `x-icon`. Just a production build getting its resources such as `runtime.js`, `polyfills.js`, `main.js`, `favicon` in its most regular way. No additional packages that should impact this behaviour, the default builder is being used.

**Comment by petebacondarwin** [comment]
I am going to transfer this to the CLI repo, since it sounds like the `<script>` tags that are being injected into the `index.html` are not containing the correct HTTP protocol...
